labs/src/main/scala/org/scalalabs/basic/lab03/PatternMatchingExercise.scala

```diff
@@ -46,8 +52,13 @@ object PatternMatchingExercise {
    *        
    *    - anything else, the function result is "Some Scala class"
    */
-  def matchOnInputType(in: Any): String = {
-    error("fix me")
+  def matchOnInputType(in: Any): String = 
+    in match{
+    case in: String => "A string with length " + in.length()
```

Comment:
For these type cases, we can also write `case _: String` for example; and the matching variable is still in scope so we can use it on the right side of the `=>`.